nms/app/views/alarms/components/hooks.ts

```diff
@@ -134,9 +178,7 @@ export function useAlertRuleReceiver({
       // remove the route
       updatedRoutes = {
         ...updatedRoutes,
-        routes: (response?.routes || []).filter(
-          route => route.receiver !== initialReceiver,
-        ),
+        routes: filterRouteByRuleName(response, initialReceiver, ruleName),
```

Comment:
Sorry, I looked at this a bit more closely. The hook and the dialog itself should probably be rewritten at some point. All of this is over complicating things. 

A given rule can only have one subscriber, we don't need the initial subscriber to update the routes. So for now we can simplify this to
```
    if (
      (receiver == null || receiver.trim() === '') &&
      initialReceiver != null
    ) {
      // remove the route
      updatedRoutes = {
        ...updatedRoutes,
        routes: (updatedRoutes.routes || []).filter(
          route => route.match?.alertname !== ruleName,
        ),
      };
    } else if (receiver != null && initialReceiver == null) {
      // creating a new route
      const newRoute: AlertRoutingTree = {
        receiver: receiver,
        match: { alertname: ruleName },
      };
      updatedRoutes = {
        ...updatedRoutes,
        routes: [...(updatedRoutes.routes || []), newRoute],
      };
    } else if (initialReceiver !== null && receiver !== null) {
      updatedRoutes = {
        ...updatedRoutes,
        routes: (updatedRoutes.routes || []).map(route => {
          if (route.match?.alertname === ruleName) {
            route.receiver = receiver;
          }
          return route;
        }),
      };
    }
```    

The tests should test against the UI then we don't have to touch them if we refactor the logic like this.